core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowIssueRequestHandler.java

```diff
@@ -185,23 +187,28 @@ static ShowIssueQuery extractQuery(ClassicHttpRequest request) {
     } catch (URISyntaxException e) {
       // Ignored
     }
-    return new ShowIssueQuery(params.get("server"), params.get("project"), params.get("issue"));
+    return new ShowIssueQuery(params.get("server"), params.get("project"), params.get("issue"), params.get("branch"), params.get("pullRequest"));
   }
 
   @VisibleForTesting
   public static class ShowIssueQuery {
     private final String serverUrl;
     private final String projectKey;
     private final String issueKey;
+    private final String branch;
+    @Nullable
+    private final String pullRequest;
 
-    public ShowIssueQuery(String serverUrl, String projectKey, String issueKey) {
+    public ShowIssueQuery(String serverUrl, String projectKey, String issueKey, String branch, @Nullable String pullRequest) {
       this.serverUrl = serverUrl;
       this.projectKey = projectKey;
       this.issueKey = issueKey;
+      this.branch = branch;
+      this.pullRequest = pullRequest;
     }
 
     public boolean isValid() {
-      return isNotBlank(serverUrl) && isNotBlank(projectKey) && isNotBlank(issueKey);
+      return isNotBlank(serverUrl) && isNotBlank(projectKey) && isNotBlank(issueKey) && isNotBlank(branch);
```

Comment:
You could also check if (pr is null or pr is not blank), to reject requests with an empty PR parameter. It would save a check later, and reject the request early

Reply:
Request without a PR is still valid, or do you mean something else?
It's only the 'branch' that will always be there

Reply:
Yes no PR is still valid, but if the value is not null, we want it to be a non-empty String (that's what you check later in the IssueApi